src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj

```
@@ -57,7 +57,7 @@
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="$(SystemDiagnosticSourceVersion)" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+  <ItemGroup Condition="$(TargetFramework) == 'net6.0' or $(TargetFramework) == 'net8.0'">
```

Sorry I missed this in my last comment, here and elsewhere, we also need single quotes around $(TargetFramework)
```suggestion
  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0' or '$(TargetFramework)' == 'net8.0'">
```